### ISSUE TITLE
[UX] Let users assign default colors to bookmark lists

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#1805]
Some use cases:

I am using different colors for bookmarks (e.g. "to visit", "favourite", "update in OSM" etc) and I would like to keep them in different lists also, but I'm too lazy to spend a few clicks more to each bookmark (just changing a color is a tedious process already!). The same goes for people who are using lists, but are to lazy to change the color as well.

My friend is using many colors for bookmarks and sometimes forgets which color is for what :) A "colored" list name will act like a note.

How it should work:

Every bookmark list has an optional color assignable by the user (no color by default).
This color is assigned by default to all new bookmarks created in the list.
If the current set is not "colored" then use the current logic of the most recently used bookmark color.
A user can change bookmark's color individually still, i.e. its possible to have bookmarks with different colors in one list.
If user changes list's assigned color then bookmarks which had the same color as list's are updated (e.g. individually colored bookmarks are not changed).

Hence people who doesn't want any behavior changes won't have any.
Other people will benefit from a more streamlined / faster bookmarks organization.

**Comment by gabriel-de-luca** [comment]
Hello, I would also like to be able to assign a color to the list of bookmarks.

I can think of a new color for the bookmarks, something like "List color". And in the lists settings, in addition to the name and description, a color which may be random by default.  

Thanks for the app.

**Comment by gabriel-de-luca** [comment]
White can be used for this new color if the bookmark isn't currently in any list.

**Comment by biodranik** [comment]
There is also another option: to make a button that changes all bookmarks colors in the list to the one specified.

**Comment by Friptick** [comment]
Seconded. Bookmark lists - i.e., types - are not especially useful if all bookmarks look the same on the map, as currently.

This project's intro mentions "travelers" and "tourists" right at top. As it happens, this is exactly the category that has been a little neglected by Osmand (in favor of outdoor sportspeople and drivers).

For traveling and tourism, decent bookmarking is super important.

Thanks for your work.

**Comment by biodranik** [comment]
You of course know that you can already change bookmark's color, right? And that any subsequently created bookmark will have the last used color. Right?

**Comment by Friptick** [comment]
Yes. We want the colors assigned _by list_. OP explains in his very first paragraph.

**Comment by biodranik** [comment]
To summarize: I see several logical and implementation issues with the proposed algorithms above.

The most important issue is that now it's clear to users that every added bookmark is 1) in the same last used set and 2) has the same last used color. Breaking that may confuse users.

A simpler solution is a button to force-change the color of all bookmarks in a list, without changing any existing logic.

**Comment by Friptick** [comment]
Alternative possibility: those users who are already assigning bookmarks to lists without any visual evidence of their action _will be happy_ to wake up and discover the bookmarks distinguished from each other appropriately.

**Comment by pastk** [comment]
> A simpler solution is a button to force-change the color of all bookmarks in a list, without changing any existing logic.

Unfortunately it seems of very little use. It'll be useful if a user decides to re-color a list of course, but that's it?
It won't help with coloring bookmarks newly added to the list, so it doesn't really solve cases from the OP.

**Comment by pastk** [comment]
> The most important issue is that now it's clear to users that every added bookmark is 1) in the same last used set and 2) has the same last used color. Breaking that may confuse users.

The proposed algo doesn't change that. A new bookmark will be added to the last used set/list and use it's color.

(There will be a change only for a rare case when the last bookmark was explicitly assigned a color different from its list.)

**Comment by pastk** [comment]
Relevant user voting stats (TG chat):

![image](https://github.com/organicmaps/organicmaps/assets/18434508/289a7efb-f638-4d97-90af-ba41a6f621dd)
(72 votes total, source: https://t.me/organicmaps/41519)

Same voting in a RU TG chat:
![image](https://github.com/organicmaps/organicmaps/assets/18434508/70a7b700-d600-46af-967e-46c017d6a96c)
(80 votes total, source: https://t.me/OrganicMapsRu/54222)

**Comment by pastk** [comment]
> Relevant user voting stats (TG chat):

It seems like shares of users (who use single color per list vs who mix several colors in a list) are very similar.
As chat users are usually more advanced/involved we may assume that there is a bigger percentage of overall users who just use a single default list with a default color or don't know its possible to change a color.

**Comment by Friptick** [comment]
Thanks @pastk for this research.

Assuming this feature gets implemented one day, there is something that could make it ten times more useful. **Use standard syntax in the underlying GPX markup!**

In concrete terms, each bookmark will probably look like `<wpt lon="XXX" lat="XX">The Place</wpt>` . Please avoid taking the OsmAnd route and hard-coding the color into each waypoint (bookmark) via proprietary tags

```
<wpt ...>
  ...
  <extensions>
    <osmand:color>#8f2511</osmand:color>
  </extensions>
</wpt>
```

Instead, just use `<type>`

```
<wpt ...>
  <type>Restaurant</type>
</wpt>
```

and make make the color of each type customizable.

That way, the bookmarks can be synced and other software (e.g., desktop or cloud) will be able to distinguish the colors too. For me, this is the holy grail that I have been waiting years for.

**Comment by biodranik** [comment]
@Friptick 

Please create a separate issue for the output GPX format so we can continue there. It is not directly related to this issue. At the moment all bookmarks are stored in KML files. KML schema has a way to assign a color to a style, and then reference this style from each object.

1. What should be the type if it is unknown (e.g. after a long tap somewhere)?
2. How the color of each type can be made customizable?

**Comment by pastk** [comment]
Do you mean 1) exactly feature types (restaurants, fuel stations, etc.) or 2) something more broad like a user-defined grouping?

If its first - then it won't work for all users, e.g. a user may want to assign different colors to different restaturants (visited, favorite, wishlist, avoid, etc).

If its 2) then it could be a list name? (but there could be format limitations to use certain symbols?)

**Comment by Friptick** [comment]
@pastk Created separate issue #8550 and added your comments there. Consider deleting them above to avoid confusion.

**Comment by biodranik** [comment]
Allowing to set all colors with a button should also solve the case with "default" colors, but in a simpler and more obvious way, right?

Then we have also tracks and their colors. Should there be a separate button to change colors of bookmarks, and a separate button to change colors of tracks?

**Comment by CedricMT** [comment]
> Allowing to set all colors with a button should also solve the case with "default" colors, but in a simpler and more obvious way, right?
> 
> Then we have also tracks and their colors. Should there be a separate button to change colors of bookmarks, and a separate button to change colors of tracks?

For me it does not solve the default list color issue. User would have to go in list and apply the change all bookmark colors option each time the new added bookmark does not have the expeted color.

**Comment by biodranik** [comment]
If it's only one bookmark, its color can be changed once for this specific bookmark. Any following bookmark will have that last used color.
If its a lot of bookmarks, all their colors can be changed to a default color once using the "set all bookmarks colors to a selected one" button.

**Comment by CedricMT** [comment]
> If it's only one bookmark, its color can be changed once for this specific bookmark. Any following bookmark will have that last used color. If its a lot of bookmarks, all their colors can be changed to a default color once using the "set all bookmarks colors to a selected one" button.

I think the initial use case was to manage multiple bookmark lists with a related color for each. So if the added bookmark does not match the latest list and color, you will have to change both every time.

**Comment by pastk** [comment]
> Allowing to set all colors with a button should also solve the case with "default" colors, but in a simpler and more obvious way, right?

You've asked this question already :)
https://github.com/organicmaps/organicmaps/issues/1805#issuecomment-2110402983

**Comment by mortenfyhn** [comment]
Just chiming in that I agree this would be great. I use OM while travelling and maintain lists for various things like attractions, birding sites, campsites etc. I currently manually change the color every time I add a new bookmark, which is quite tedious, especially with the new more advanced color picker.

**Comment by mortenfyhn** [comment]
Hi all again.

I've implemented a variant of this in my fork. It works like this:

1. Like bookmarks, lists can also have a color. Lists without a color assigned behave like today.
2. "Change color of all bookmarks" does what it says like before, but now also assigns that color to the list
3. New bookmarks added to the list will then get that color, not the global last used color
4. Individual bookmarks colors can still be changed, so you can override the list color if any
5.  Imported bookmarks keep their original colors
6. The show/hide 👁️ symbol next to each list shows the list's color

I think this mostly matches spec suggestions above here.

Here are the changes:

* mortenfyhn/organicmaps@a434e0c111
* mortenfyhn/organicmaps@1b9ca1a1df
* mortenfyhn/organicmaps@375db53db9

I'd be happy to PR this if there's interest. It's AI-coded, but I can go thru it myself first.

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/2eec67c7-9f64-4529-9e1d-9ebc1a3e3a86" />

**Comment by biodranik** [comment]
One of the arguments against "default list colors" is that some lists will have it, and some won't, and users will need to remember which ones.

With the current implementation, it is always one explicit user action to recolor, and the last-used color is correctly applied for any newly added bookmarks going into the same list, without "default" cognitive complexity.

**Comment by mortenfyhn** [comment]
That's a good point. I think "user has to remember" is a central challenge to both sides of this. You shouldn't have to remember the color of a bookmark or the color of a list.

> With the current implementation, it is always one explicit user action to recolor, and the last-used color is correctly applied for any newly added bookmarks going into the same list, without "default" cognitive complexity.

My challenge with this is that I typically add say a few bookmarks to different lists, and then I have to re-color each one to match the list color I want, and I need to **remember** the color of my list. ("Which of the four blue color presets was it again?")

Perhaps something like this would help, it shows the default color of a list *if it has one*:

<img width="400"  alt="Image" src="https://github.com/user-attachments/assets/8478021f-d186-40aa-b312-6c62f1935185" />
